### PR TITLE
Add custom marshaller tests for delegates

### PR DESCRIPTION
**Pull request by kant2002**
These tests added to better test ICustomMarshaller support in NativeAOT as per discussed with @jkotas

### CODE REVIEW

**Review by jkoritzinsky — APPROVED** [review]
Thanks for improving our test coverage!

**Comment by kant2002** [comment]
@jkoritzinsky anything else needed for this PR?

**Comment by jkoritzinsky** [comment]
Can you remove the places where you set CallingConvention? In native code, STDMETHODCALLTYPE is stdcall on Windows and cdecl on non-Windows, aka the DllImport default.

File: src/tests/Interop/ICustomMarshaler/Primitives/ICustomMarshalerNative.cpp [hunk 1]
```diff
@@ -7,27 +7,27 @@
 using TestDelegate = LPCSTR(STDMETHODCALLTYPE*)(int);
 using TestDelegateRef = LPCSTR(STDMETHODCALLTYPE*)(int*);
 
-extern "C" int DLL_EXPORT STDMETHODCALLTYPE NativeParseInt(LPCSTR str)
```
Comment (jkoritzinsky) [inline src/tests/Interop/ICustomMarshaler/Primitives/ICustomMarshalerNative.cpp]: Please keep the `STDMETHODCALLTYPE`s on the native methods without adding a `CallingConvention=` setter in managed.

**Review by jkoritzinsky — CHANGES_REQUESTED** [review]

File: src/tests/Interop/ICustomMarshaler/Primitives/ICustomMarshalerNative.cpp (hunk 1 above)
Comment (kant2002) [reply src/tests/Interop/ICustomMarshaler/Primitives/ICustomMarshalerNative.cpp]: Yeah. I reread again and understand that you want like that. Done.

**Review by jkoritzinsky — APPROVED** [review]

**Comment by kant2002** [comment]
@jkoritzinsky can you merge now? I would like to continue work on ICustomMarshaller for NativeAOT

**Pull request merged**